### PR TITLE
DDF-2917 (2.10.x) Update caching to exclude base app pages and html

### PR DESCRIPTION
**Pull request by andrewkfiedler**
#### What does this PR do?
 - As of DDF-2893, we're caching everything far too aggressively.
 - This update allows apps to perform their own cache busting between versions, see DDF-2916.
 - Without this, a user will need to clear their cache (or wait a week) when upgrading to new versions.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@andrewkfiedler
@djblue 
@rzwiefel 
@kcwire
@pklinef

#### How should this be tested? (List steps with links to updated documentation)
Go to the admin application.  Verify that the main app page and any html pages within do not get cached.  Verify that other assets do.

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2917

### CODE REVIEW

File: platform/platform-filter-response/src/main/java/org/codice/ddf/platform/response/filter/ResponseFilter.java [hunk 1]
```diff
@@ -59,16 +60,17 @@
             X_CONTENT_SECURITY_POLICY + "=" + DEFAULT_CONTENT_SECURITY_POLICY,
             CACHE_CONTROL + "=" + DEFAULT_CACHE_CONTROL_VALUE);
 
-    @Override
-    public void init(FilterConfig filterConfig) throws ServletException {
-        LOGGER.debug("Initializing Response Security Filter.");
+    // Index paths (such as /search/catalog/ or /admin/) should never be cached
+    private void disableCachingForHTML(HttpServletRequest request, HttpServletResponse response) {
```
Comment (brendan-hofmann) [inline platform/platform-filter-response/src/main/java/org/codice/ddf/platform/response/filter/ResponseFilter.java]: I think this invalidates the comment in the pom. 😄 We should unit test each case handled here.

We should also probably update the documentation for this component, to reflect that we automatically disable caching for html and index paths, regardless of what is specified in the configuration.

Comment (andrewkfiedler) [inline platform/platform-filter-response/src/main/java/org/codice/ddf/platform/response/filter/ResponseFilter.java]: Sounds good, can you work with me on this update?  Trying to get it in for the 2.10.2 release.

Comment (pklinef) [inline platform/platform-filter-response/src/main/java/org/codice/ddf/platform/response/filter/ResponseFilter.java]: I agree that in general this should be configurable but that can maybe wait to be done on master with #1841 so this is at least fixed for 2.10.2.

Comment (brendan-hofmann) [inline platform/platform-filter-response/src/main/java/org/codice/ddf/platform/response/filter/ResponseFilter.java]: Given the upcoming release I agree those can wait for the master PR, like @pklinef said.

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
Heroed.
Confirmed cache busting works on all root pages and html pages.

**Pull request merged**